ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -177,6 +179,9 @@ public void onOpen(final Session session, final EndpointConfig unused) {
       return;
     }
     try {
+      // Check if the user has authorization to access this Websocket endpoint
+      checkEndpointAuthorization(session.getUserPrincipal(), "onOpen");
```

Comment:
hard-coding `onOpen` is very fragile. does it refer to the method name? the annotation on the method name? some magic constant? If I were a user implementing this, how would I know what to implement? as it stands, I need to examine the code?

Reply:
It's the method name. Any idea how to get it without hard-coding it?

Reply:
The problem is that perhaps not hard-coding it is even more fragile :D I think a constant would be good enough for now

Reply:
Done. I used a constant.